src/server/transaction.cc

```diff
@@ -484,14 +484,16 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
       local_result_ = status;
     } else {
       if (status == OpStatus::OUT_OF_MEMORY) {
+        absl::base_internal::SpinLockHolder lk{&local_result_mu_};
+        CHECK(local_result_ == OpStatus::OK || local_result_ == OpStatus::OUT_OF_MEMORY);
         local_result_ = status;
       } else {
         CHECK_EQ(OpStatus::OK, status);
       }
     }
   } catch (std::bad_alloc&) {
-    // TODO: to log at most once per sec.
-    LOG_FIRST_N(ERROR, 16) << " out of memory";
+    LOG_FIRST_N(ERROR, 16) << " out of memory";  // TODO: to log at most once per sec.
```

Comment:
How about `LOG_EVERY_N_SEC(ERROR, 600)` (or any other value)?

Reply:
We don't have it for some reason, but upstream absl has it 🤔 I guess we need to update deps

Reply:
It might be because we're using glog directly and not through absl, at least that's something I remember seeing?

Reply:
Yes, let's fix it in a separate PR

Reply:
no, no please do not fix it.